From d45723648d84ef30b3a9fe3d9a78f9cf688bc6bb Mon Sep 17 00:00:00 2001
From: Mark Pariente <markpariente@gmail.com>
Date: Sat, 19 Oct 2013 19:20:13 -0700
Subject: [PATCH] Fix core/window.c compile with GCC 4.8 and -O3

Using GCC 4.8.1 with -O3 results in compile failure (*). This commit fixes
the issue by initializing x and y in the default case of warp_grab_pointer().
Also added a warning to help with extensions calling into
meta_window_begin_grab_op() for something other than keyboard grab ops.

[*] Without fix:

 core/window.c: In function 'meta_window_begin_grab_op':
 core/window.c:10274:30: error: 'y' may be used uninitialized in this function
 [-Werror=maybe-uninitialized]
    meta_display_begin_grab_op (window->display,
                               ^
 core/window.c:10274:30: error: 'x' may be used uninitialized in this function
 [-Werror=maybe-uninitialized]
 cc1: all warnings being treated as errors
---
 src/core/window.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/core/window.c b/src/core/window.c
index f5826bb..08577de 100644
--- a/src/core/window.c
+++ b/src/core/window.c
@@ -10211,6 +10211,9 @@ warp_grab_pointer (MetaWindow          *window,
         break;
 
       default:
+        g_warning ("meta_window_begin_grab_op() can only be called for keyboard grab ops");
+        *x = 0;
+        *y = 0;
         return FALSE;
     }
 
-- 
1.8.4

